# ######Women Commission Complaints in Maharashtra: A Data Analysis

## Introduction

This notebook analyses data on complaints received and handled by the Maharashtra State Commission for Women.

The dataset contains information on different categories of complaints, including marital problems, social problems and rape, property issues, sexual harassment at the workplace, harassment at the workplace and other complaints.

The aim of this analysis is to identify patterns in the types of complaints received, the number of complaints resolved, and the complaints that remain under proceedings.

In [1]:
import pandas as pd

In [2]:
df = pd.read_excel("Women_Commission_ST_Mahamandal.xlsx", sheet_name="Sheet1", header=None)

In [3]:
df.shape

(123, 13)

In [4]:
df.head(20)

,0,1,2,3,4,5,6,7,8,9,10,11,12
0,NaN,NaN,1 January 2021 - 31 March 2021,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,A,B,(A + B),NaN,NaN,NaN,NaN,NaN,2021-01-01 00:00:00,NaN,NaN,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,Maratial problems,1877.333333,455,6087,408,5679,NaN,NaN,A,B,(A + B),NaN,NaN
6,Pending court claims,0,0,0,0,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
7,Dowry claims,0,0,0,0,0,NaN,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
8,"Social problems, rape etc.",1185,453,4008,432,3576,NaN,Maratial problems,1854,149,2003,134,1869
9,Property issues,287.333333,146,1008,139,869,NaN,Pending court claims,0,0,0,0,0


### Inspecting the Raw Data

I will now examine the rows and columns of the raw dataset to identify the different sections, headers and categories present in the spreadsheet.

In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 123 entries, 0 to 122
Data columns (total 13 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       50 non-null     str   
 1   1       55 non-null     object
 2   2       60 non-null     object
 3   3       55 non-null     object
 4   4       70 non-null     object
 5   5       72 non-null     object
 6   6       24 non-null     object
 7   7       52 non-null     object
 8   8       53 non-null     object
 9   9       56 non-null     object
 10  10      33 non-null     object
 11  11      30 non-null     object
 12  12      29 non-null     object
dtypes: object(12), str(1)
memory usage: 12.6+ KB


In [6]:
df.isnull().sum()

0     73
1     68
2     63
3     68
4     53
5     51
6     99
7     71
8     70
9     67
10    90
11    93
12    94
dtype: int64

### Looking at the Raw Spreadsheet

The dataset contains blank cells and multiple header rows. I will inspect the raw values to determine how the information is organised before restructuring it for analysis.

In [7]:
pd.set_option('display.max_rows', 150)
pd.set_option('display.max_columns', 20)

df

,0,1,2,3,4,5,6,7,8,9,10,11,12
0,NaN,NaN,1 January 2021 - 31 March 2021,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,A,B,(A + B),NaN,NaN,NaN,NaN,NaN,2021-01-01 00:00:00,NaN,NaN,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,Maratial problems,1877.333333,455,6087,408,5679,NaN,NaN,A,B,(A + B),NaN,NaN
6,Pending court claims,0,0,0,0,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
7,Dowry claims,0,0,0,0,0,NaN,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
8,"Social problems, rape etc.",1185,453,4008,432,3576,NaN,Maratial problems,1854,149,2003,134,1869
9,Property issues,287.333333,146,1008,139,869,NaN,Pending court claims,0,0,0,0,0


## 2. Separating the Data Sections

The original spreadsheet contains two related tables placed next to each other. The first section contains information on complaints received and their status, while the second section contains category-wise complaint data.

To make the data easier to analyse, the two sections will be separated into individual dataframes.

In [8]:

df_left = df.iloc[:, 0:6]


df_right = df.iloc[:, 7:13]

df_left.head(10)

,0,1,2,3,4,5
0,NaN,NaN,1 January 2021 - 31 March 2021,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,A,B,(A + B),NaN,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869


In [9]:
df_right.head(10)

,7,8,9,10,11,12
0,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,NaN,2021-01-01 00:00:00,NaN,NaN,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN
4,NaN,NaN,NaN,NaN,NaN,NaN
5,NaN,A,B,(A + B),NaN,NaN
6,NaN,NaN,NaN,NaN,NaN,NaN
7,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
8,Maratial problems,1854,149,2003,134,1869
9,Pending court claims,0,0,0,0,0


### Identifying the Table Header

The raw spreadsheet contains several rows of headings and notes before the actual complaint records begin. I will identify the row containing the meaningful column names so that the dataset can be cleaned for analysis.

In [10]:
df.iloc[4:15, 0:6]

,0,1,2,3,4,5
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434


In [11]:
df.iloc[4:15, 7:13]

,7,8,9,10,11,12
4,NaN,NaN,NaN,NaN,NaN,NaN
5,NaN,A,B,(A + B),NaN,NaN
6,NaN,NaN,NaN,NaN,NaN,NaN
7,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
8,Maratial problems,1854,149,2003,134,1869
9,Pending court claims,0,0,0,0,0
10,Dowry claims,0,0,0,0,0
11,"Social problems, rape etc.",1171,151,1322,142,1180
12,Property issues,286,46,332,44,288
13,Sexual harassment at work place,36,1,37,4,33


## 3. Identifying the Data Records

The raw spreadsheet contains multiple sections of complaint data. Before cleaning the dataset, I will identify the rows that contain actual complaint categories and distinguish them from blank rows and headings.

In [12]:
df[df[0].notna()][[0, 1, 2, 3, 4, 5]]

,0,1,2,3,4,5
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434


In [13]:
df[df[7].notna()][[7, 8, 9, 10, 11, 12]]

,7,8,9,10,11,12
7,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
8,Maratial problems,1854,149,2003,134,1869
9,Pending court claims,0,0,0,0,0
10,Dowry claims,0,0,0,0,0
11,"Social problems, rape etc.",1171,151,1322,142,1180
12,Property issues,286,46,332,44,288
13,Sexual harassment at work place,36,1,37,4,33
14,Harassment at work place,392,43,435,52,383
15,Others,1008,141,1149,131,1018
16,Total,4747,531,5278,507,4771


### Identifying the Reporting Periods

The dataset appears to contain complaint figures for different reporting periods. Identifying these periods will allow the cleaned dataset to retain a time dimension for later analysis.

In [14]:
df.iloc[:, :13].head(40)

,0,1,2,3,4,5,6,7,8,9,10,11,12
0,NaN,NaN,1 January 2021 - 31 March 2021,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,A,B,(A + B),NaN,NaN,NaN,NaN,NaN,2021-01-01 00:00:00,NaN,NaN,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,Maratial problems,1877.333333,455,6087,408,5679,NaN,NaN,A,B,(A + B),NaN,NaN
6,Pending court claims,0,0,0,0,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
7,Dowry claims,0,0,0,0,0,NaN,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
8,"Social problems, rape etc.",1185,453,4008,432,3576,NaN,Maratial problems,1854,149,2003,134,1869
9,Property issues,287.333333,146,1008,139,869,NaN,Pending court claims,0,0,0,0,0


In [15]:
df[df[0].notna()][[0, 1, 2, 3, 4, 5]]

,0,1,2,3,4,5
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434


In [16]:
df[df[7].notna()][[7, 8, 9, 10, 11, 12]]

,7,8,9,10,11,12
7,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
8,Maratial problems,1854,149,2003,134,1869
9,Pending court claims,0,0,0,0,0
10,Dowry claims,0,0,0,0,0
11,"Social problems, rape etc.",1171,151,1322,142,1180
12,Property issues,286,46,332,44,288
13,Sexual harassment at work place,36,1,37,4,33
14,Harassment at work place,392,43,435,52,383
15,Others,1008,141,1149,131,1018
16,Total,4747,531,5278,507,4771


In [17]:
df.iloc[:, :13].head(40)

,0,1,2,3,4,5,6,7,8,9,10,11,12
0,NaN,NaN,1 January 2021 - 31 March 2021,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,A,B,(A + B),NaN,NaN,NaN,NaN,NaN,2021-01-01 00:00:00,NaN,NaN,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,Maratial problems,1877.333333,455,6087,408,5679,NaN,NaN,A,B,(A + B),NaN,NaN
6,Pending court claims,0,0,0,0,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN
7,Dowry claims,0,0,0,0,0,NaN,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
8,"Social problems, rape etc.",1185,453,4008,432,3576,NaN,Maratial problems,1854,149,2003,134,1869
9,Property issues,287.333333,146,1008,139,869,NaN,Pending court claims,0,0,0,0,0


## 4. Identifying Reporting Periods

The spreadsheet contains multiple blocks of complaint data covering different reporting periods. Each block follows a similar structure, with complaint categories followed by figures for pending, registered, total, resolved and ongoing complaints.

I will identify the reporting-period markers before restructuring the data.

In [18]:
date_rows = df[df.apply(
    lambda row: row.astype(str).str.contains("2021|2022|2023|2024|2025", regex=True).any(),
    axis=1
)]

date_rows

,0,1,2,3,4,5,6,7,8,9,10,11,12
0,NaN,NaN,1 January 2021 - 31 March 2021,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,A,B,(A + B),NaN,NaN,NaN,NaN,NaN,2021-01-01 00:00:00,NaN,NaN,NaN
16,NaN,NaN,1 April 2021 - 31 March 2022,NaN,NaN,NaN,NaN,Total,4747,531,5278,507,4771
20,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings,NaN,NaN,NaN,2021-02-01 00:00:00,NaN,NaN,NaN
32,NaN,NaN,1 April 2022 - 31 March 2023,NaN,NaN,NaN,NaN,Others,1018,108,1126,103,1023
37,Maratial problems,2374,2819,5193,2841,2352,NaN,NaN,NaN,2021-03-01 00:00:00,NaN,NaN,NaN
48,NaN,NaN,1 April 2023 - 31 March 2024,NaN,NaN,NaN,NaN,Harassment at work place,387,35,422,41,381
64,NaN,NaN,1 April 2024 - 31 October 2024,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
82,NaN,NaN,NaN,NaN,NaN,NaN,1 January 2021 - 31 October 2024,NaN,NaN,NaN,NaN,NaN,NaN
109,NaN,NaN,NaN,NaN,NaN,NaN,1 January 2021 - 31 October 2024,NaN,NaN,NaN,NaN,NaN,NaN


## 5. Cleaning the Complaint Data

The first table contains category-wise information on complaints received by the commission. It records the number of old pending complaints, newly registered complaints, total complaints, resolved complaints and complaints under proceedings.

I will extract the relevant rows and columns from the raw spreadsheet and remove the formatting rows that are not part of the actual data.

In [19]:
left = df.iloc[:, 0:6].copy()

left.head(15)

,0,1,2,3,4,5
0,NaN,NaN,1 January 2021 - 31 March 2021,NaN,NaN,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,A,B,(A + B),NaN,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869


### Creating Meaningful Column Names

The fourth row of the raw spreadsheet contains the column names for the complaint data. I will use these names to create a cleaner dataframe.

In [20]:
left.columns = left.iloc[4]
left = left.iloc[5:].copy()

left.head(10)

4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434
14,NaN,NaN,NaN,NaN,NaN,NaN


### Renaming the Columns

The first column contains the complaint categories, so I will rename it to `category`. The remaining columns will also be given shorter names to make analysis easier.

In [21]:
left.columns = [
    "category",
    "old_pending",
    "registered",
    "total_complaints",
    "resolved",
    "under_proceedings"
]

left.head(10)

,category,old_pending,registered,total_complaints,resolved,under_proceedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434
14,NaN,NaN,NaN,NaN,NaN,NaN


In [22]:
left = left.dropna(how="all")

left

,category,old_pending,registered,total_complaints,resolved,under_proceedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434
16,NaN,NaN,1 April 2021 - 31 March 2022,NaN,NaN,NaN


### Removing the Summary Row

The spreadsheet includes a `Total` row that summarises all complaint categories. Since the analysis will calculate totals and proportions directly from the individual categories, this summary row is excluded to avoid double counting.

In [23]:
left = left[left["category"].astype(str).str.strip() != "Total"]

left

,category,old_pending,registered,total_complaints,resolved,under_proceedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
16,NaN,NaN,1 April 2021 - 31 March 2022,NaN,NaN,NaN
18,NaN,A,B,(A + B),NaN,NaN


In [24]:
period1 = df.iloc[4:14, 0:6].copy()

period1

,0,1,2,3,4,5
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434


### Standardising Column Names

The first row contains the original column headings. I will replace these with shorter, descriptive names that are easier to use during analysis.


In [25]:
period1.columns = [
    "category",
    "old_pending",
    "registered",
    "total_complaints",
    "resolved",
    "under_proceedings"
]

period1

,category,old_pending,registered,total_complaints,resolved,under_proceedings
4,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434


### Checking Data Types

Before analysing the data, I will check how Python has interpreted each column. This helps identify whether the complaint figures are stored as numbers and whether any cleaning is required.

In [26]:
period1.dtypes

category                str
old_pending          object
registered           object
total_complaints     object
resolved             object
under_proceedings    object
dtype: object

In [27]:
period1.isnull().sum()

category             0
old_pending          0
registered           0
total_complaints     0
resolved             0
under_proceedings    0
dtype: int64

### Removing the Original Header Row

The original spreadsheet contains a second header row inside the data. Since the column names have already been standardised, this row is removed before converting the numerical fields into numeric data types.

In [28]:
period1 = period1.iloc[1:].copy()

period1

,category,old_pending,registered,total_complaints,resolved,under_proceedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0,0,0,0,0
7,Dowry claims,0,0,0,0,0
8,"Social problems, rape etc.",1185,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434


In [29]:

numeric_columns = [
    "old_pending",
    "registered",
    "total_complaints",
    "resolved",
    "under_proceedings"
]

for col in numeric_columns:
    period1[col] = pd.to_numeric(period1[col], errors="coerce")

period1.dtypes

category                 str
old_pending          float64
registered             int64
total_complaints       int64
resolved               int64
under_proceedings      int64
dtype: object

In [30]:
period1

,category,old_pending,registered,total_complaints,resolved,under_proceedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0.000000,0,0,0,0
7,Dowry claims,0.000000,0,0,0,0
8,"Social problems, rape etc.",1185.000000,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35.000000,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057
13,Total,4791.666667,1527,15892,1428,14434


## 6. Removing the Total Row

The dataset contains a `Total` row that summarises all complaint categories. Since we will calculate our own totals and comparisons from the individual categories, this row is removed to avoid double-counting.

In [31]:
period1 = period1[period1["category"].str.strip() != "Total"].copy()

period1

,category,old_pending,registered,total_complaints,resolved,under_proceedings
5,Maratial problems,1877.333333,455,6087,408,5679
6,Pending court claims,0.000000,0,0,0,0
7,Dowry claims,0.000000,0,0,0,0
8,"Social problems, rape etc.",1185.000000,453,4008,432,3576
9,Property issues,287.333333,146,1008,139,869
10,Sexual harassment at work place,35.000000,8,113,11,102
11,Harassment at work place,387.333333,108,1270,119,1151
12,Others,1016.333333,357,3406,349,3057


## 7. Complaint Categories

I will compare the total number of complaints across categories to identify which types of complaints account for the largest share of cases.

In [32]:
period1[["category", "total_complaints"]].sort_values(
    "total_complaints",
    ascending=False
)

,category,total_complaints
5,Maratial problems,6087
8,"Social problems, rape etc.",4008
12,Others,3406
11,Harassment at work place,1270
9,Property issues,1008
10,Sexual harassment at work place,113
7,Dowry claims,0
6,Pending court claims,0


## 8. Complaint Resolution Rate

The number of complaints registered does not show how effectively complaints were resolved. I will therefore calculate the proportion of total complaints that were resolved in each category.

In [33]:
period1["resolution_rate"] = (
    period1["resolved"] / period1["total_complaints"] * 100
)

period1[[
    "category",
    "total_complaints",
    "resolved",
    "resolution_rate"
]].sort_values(
    "resolution_rate",
    ascending=False
)

,category,total_complaints,resolved,resolution_rate
9,Property issues,1008,139,13.789683
8,"Social problems, rape etc.",4008,432,10.778443
12,Others,3406,349,10.246624
10,Sexual harassment at work place,113,11,9.734513
11,Harassment at work place,1270,119,9.370079
5,Maratial problems,6087,408,6.702809
6,Pending court claims,0,0,NaN
7,Dowry claims,0,0,NaN


## 9. Which Complaints Are Most Common?

To identify the main areas of concern handled by the commission, I will rank complaint categories by the total number of complaints recorded.

In [34]:
top_categories = period1[
    ["category", "total_complaints"]
].sort_values(
    "total_complaints",
    ascending=False
)

top_categories

,category,total_complaints
5,Maratial problems,6087
8,"Social problems, rape etc.",4008
12,Others,3406
11,Harassment at work place,1270
9,Property issues,1008
10,Sexual harassment at work place,113
7,Dowry claims,0
6,Pending court claims,0


### Complaints by Category

The following chart compares the total number of complaints recorded across different categories.

In [35]:
top_categories

,category,total_complaints
5,Maratial problems,6087
8,"Social problems, rape etc.",4008
12,Others,3406
11,Harassment at work place,1270
9,Property issues,1008
10,Sexual harassment at work place,113
7,Dowry claims,0
6,Pending court claims,0


In [36]:
period1[
    ["category", "resolved", "under_proceedings"]
].sort_values(
    "under_proceedings",
    ascending=False
)

,category,resolved,under_proceedings
5,Maratial problems,408,5679
8,"Social problems, rape etc.",432,3576
12,Others,349,3057
11,Harassment at work place,119,1151
9,Property issues,139,869
10,Sexual harassment at work place,11,102
7,Dowry claims,0,0
6,Pending court claims,0,0


In [37]:
top_categories

,category,total_complaints
5,Maratial problems,6087
8,"Social problems, rape etc.",4008
12,Others,3406
11,Harassment at work place,1270
9,Property issues,1008
10,Sexual harassment at work place,113
7,Dowry claims,0
6,Pending court claims,0


In [38]:
period1[
    ["category", "resolved", "under_proceedings"]
].sort_values(
    "under_proceedings",
    ascending=False
)

,category,resolved,under_proceedings
5,Maratial problems,408,5679
8,"Social problems, rape etc.",432,3576
12,Others,349,3057
11,Harassment at work place,119,1151
9,Property issues,139,869
10,Sexual harassment at work place,11,102
7,Dowry claims,0,0
6,Pending court claims,0,0


In [39]:
period1["resolution_rate"] = (
    period1["resolved"] / period1["total_complaints"] * 100
)

period1[
    ["category", "total_complaints", "resolved", "resolution_rate"]
].sort_values(
    "resolution_rate",
    ascending=False
)

,category,total_complaints,resolved,resolution_rate
9,Property issues,1008,139,13.789683
8,"Social problems, rape etc.",4008,432,10.778443
12,Others,3406,349,10.246624
10,Sexual harassment at work place,113,11,9.734513
11,Harassment at work place,1270,119,9.370079
5,Maratial problems,6087,408,6.702809
6,Pending court claims,0,0,NaN
7,Dowry claims,0,0,NaN


In [40]:
period1["proceeding_rate"] = (
    period1["under_proceedings"] / period1["total_complaints"] * 100
)

period1[
    ["category", "total_complaints", "under_proceedings", "proceeding_rate"]
].sort_values(
    "proceeding_rate",
    ascending=False
)

,category,total_complaints,under_proceedings,proceeding_rate
5,Maratial problems,6087,5679,93.297191
11,Harassment at work place,1270,1151,90.629921
10,Sexual harassment at work place,113,102,90.265487
12,Others,3406,3057,89.753376
8,"Social problems, rape etc.",4008,3576,89.221557
9,Property issues,1008,869,86.210317
6,Pending court claims,0,0,NaN
7,Dowry claims,0,0,NaN


In [41]:
total_complaints = period1["total_complaints"].sum()
total_resolved = period1["resolved"].sum()
total_under_proceedings = period1["under_proceedings"].sum()

overall_resolution_rate = (total_resolved / total_complaints) * 100
overall_proceeding_rate = (total_under_proceedings / total_complaints) * 100

print("Total complaints:", total_complaints)
print("Total resolved:", total_resolved)
print("Total under proceedings:", total_under_proceedings)
print("Overall resolution rate:", overall_resolution_rate)
print("Overall proceeding rate:", overall_proceeding_rate)

Total complaints: 15892
Total resolved: 1458
Total under proceedings: 14434
Overall resolution rate: 9.174427384847721
Overall proceeding rate: 90.82557261515227


In [42]:
period1[["category", "total_complaints", "resolved", "under_proceedings"]]

,category,total_complaints,resolved,under_proceedings
5,Maratial problems,6087,408,5679
6,Pending court claims,0,0,0
7,Dowry claims,0,0,0
8,"Social problems, rape etc.",4008,432,3576
9,Property issues,1008,139,869
10,Sexual harassment at work place,113,11,102
11,Harassment at work place,1270,119,1151
12,Others,3406,349,3057


In [43]:
period1[["category", "total_complaints", "resolved", "under_proceedings"]].sort_values(
    "total_complaints",
    ascending=False
)

,category,total_complaints,resolved,under_proceedings
5,Maratial problems,6087,408,5679
8,"Social problems, rape etc.",4008,432,3576
12,Others,3406,349,3057
11,Harassment at work place,1270,119,1151
9,Property issues,1008,139,869
10,Sexual harassment at work place,113,11,102
7,Dowry claims,0,0,0
6,Pending court claims,0,0,0


## 11. Concentration of Complaints

The distribution of complaints is highly concentrated across a small number of categories. I will calculate the percentage of total complaints represented by each category to understand how much of the overall caseload is concentrated in the largest complaint types.

In [44]:
period1["share_of_complaints"] = (
    period1["total_complaints"] / total_complaints * 100
)

period1[
    ["category", "total_complaints", "share_of_complaints"]
].sort_values(

    "share_of_complaints",
    ascending=False
)

,category,total_complaints,share_of_complaints
5,Maratial problems,6087,38.302290
8,"Social problems, rape etc.",4008,25.220237
12,Others,3406,21.432167
11,Harassment at work place,1270,7.991442
9,Property issues,1008,6.342814
10,Sexual harassment at work place,113,0.711050
7,Dowry claims,0,0.000000
6,Pending court claims,0,0.000000


In [45]:
top_three_share = (
    period1.nlargest(3, "total_complaints")["total_complaints"].sum()
    / total_complaints * 100
)

print("Share of complaints from the top three categories:", top_three_share)

Share of complaints from the top three categories: 84.95469418575384


## 12. Comparing Reporting Periods

The dataset contains complaint figures for multiple reporting periods. After examining the first period, I will extract the next period using the same structure so that complaint volumes and resolution patterns can eventually be compared over time.

In [46]:
period2 = df.iloc[36:46, 0:6].copy()

period2

,0,1,2,3,4,5
36,Column 1,Total old pending complaints,Total complaints registered,Total complaints,Resolved complaints,Compaints under procedings
37,Maratial problems,2374,2819,5193,2841,2352
38,Pending court claims,0,0,0,0,0
39,Dowry claims,0,0,0,0,0
40,"Social problems, rape etc.",1659,2803,4462,3020,1442
41,Property issues,323,909,1232,888,344
42,Sexual harassment at work place,33,37,70,20,50
43,Harassment at work place,402,548,950,525,425
44,Others,1075,3030,4105,2878,1227
45,Total,5866,10146,16012,10172,5840


### Cleaning the Second Reporting Period

The second reporting block covers 1 April 2021 to 31 March 2022. I will apply the same cleaning process used for the first reporting period so that the two periods can be compared consistently.

In [47]:
period2.columns = [
    "category",
    "old_pending",
    "registered",
    "total_complaints",
    "resolved",
    "under_proceedings"
]

period2 = period2.iloc[1:].copy()

period2 = period2[
    period2["category"].str.strip() != "Total"
].copy()

numeric_columns = [
    "old_pending",
    "registered",
    "total_complaints",
    "resolved",
    "under_proceedings"
]

for col in numeric_columns:
    period2[col] = pd.to_numeric(period2[col], errors="coerce")

period2

,category,old_pending,registered,total_complaints,resolved,under_proceedings
37,Maratial problems,2374,2819,5193,2841,2352
38,Pending court claims,0,0,0,0,0
39,Dowry claims,0,0,0,0,0
40,"Social problems, rape etc.",1659,2803,4462,3020,1442
41,Property issues,323,909,1232,888,344
42,Sexual harassment at work place,33,37,70,20,50
43,Harassment at work place,402,548,950,525,425
44,Others,1075,3030,4105,2878,1227


### Removing Empty Categories

Pending court claims and dowry claims contain zero complaints in this reporting period. These categories are excluded from the analysis because they contain no recorded cases.

In [48]:
period2 = period2[
    period2["total_complaints"] > 0
].copy()

period2

,category,old_pending,registered,total_complaints,resolved,under_proceedings
37,Maratial problems,2374,2819,5193,2841,2352
40,"Social problems, rape etc.",1659,2803,4462,3020,1442
41,Property issues,323,909,1232,888,344
42,Sexual harassment at work place,33,37,70,20,50
43,Harassment at work place,402,548,950,525,425
44,Others,1075,3030,4105,2878,1227


## 13. Overall Complaint Outcomes

I will calculate the total number of complaints, resolved complaints and complaints under proceedings for the second reporting period.

In [49]:
total_complaints_2 = period2["total_complaints"].sum()
total_resolved_2 = period2["resolved"].sum()
total_proceedings_2 = period2["under_proceedings"].sum()

resolution_rate_2 = (
    total_resolved_2 / total_complaints_2 * 100
)

proceeding_rate_2 = (
    total_proceedings_2 / total_complaints_2 * 100
)

print("Total complaints:", total_complaints_2)
print("Total resolved:", total_resolved_2)
print("Total under proceedings:", total_proceedings_2)
print("Resolution rate:", resolution_rate_2)
print("Proceeding rate:", proceeding_rate_2)

Total complaints: 16012
Total resolved: 10172
Total under proceedings: 5840
Resolution rate: 63.5273544841369
Proceeding rate: 36.4726455158631


## 14. Comparing Complaint Outcomes Across Periods

Overall resolution rates changed substantially between the two reporting periods. I will now compare individual complaint categories to determine whether the change was broad-based or concentrated in particular types of complaints.

In [50]:
comparison = period1[
    ["category", "total_complaints", "resolved", "under_proceedings"]
].merge(
    period2[
        ["category", "total_complaints", "resolved", "under_proceedings"]
    ],
    on="category",
    suffixes=("_period1", "_period2")
)

comparison

,category,total_complaints_period1,resolved_period1,under_proceedings_period1,total_complaints_period2,resolved_period2,under_proceedings_period2
0,Maratial problems,6087,408,5679,5193,2841,2352
1,"Social problems, rape etc.",4008,432,3576,4462,3020,1442
2,Property issues,1008,139,869,1232,888,344
3,Sexual harassment at work place,113,11,102,70,20,50
4,Harassment at work place,1270,119,1151,950,525,425
5,Others,3406,349,3057,4105,2878,1227


In [51]:
comparison["resolution_rate_period1"] = (
    comparison["resolved_period1"] /
    comparison["total_complaints_period1"] * 100
)

comparison["resolution_rate_period2"] = (
    comparison["resolved_period2"] /
    comparison["total_complaints_period2"] * 100
)

comparison[
    [
        "category",
        "resolution_rate_period1",
        "resolution_rate_period2"
    ]
]

,category,resolution_rate_period1,resolution_rate_period2
0,Maratial problems,6.702809,54.708261
1,"Social problems, rape etc.",10.778443,67.682654
2,Property issues,13.789683,72.077922
3,Sexual harassment at work place,9.734513,28.571429
4,Harassment at work place,9.370079,55.263158
5,Others,10.246624,70.109622


## 15. Change in Complaint Volume

Resolution rates can be affected by the length of the reporting period. I will therefore compare the number of complaints registered in each period alongside the resolution figures.

In [52]:
print("Period 1 registered:", period1["registered"].sum())
print("Period 2 registered:", period2["registered"].sum())

change = (
    (period2["registered"].sum() - period1["registered"].sum())
    / period1["registered"].sum()
) * 100

print("Change in registered complaints:", change, "%")

Period 1 registered: 1527
Period 2 registered: 10146
Change in registered complaints: 564.4400785854617 %


In [53]:
period1[["category", "registered", "resolved"]]

,category,registered,resolved
5,Maratial problems,455,408
6,Pending court claims,0,0
7,Dowry claims,0,0
8,"Social problems, rape etc.",453,432
9,Property issues,146,139
10,Sexual harassment at work place,8,11
11,Harassment at work place,108,119
12,Others,357,349


In [54]:
period2[["category", "registered", "resolved"]]

,category,registered,resolved
37,Maratial problems,2819,2841
40,"Social problems, rape etc.",2803,3020
41,Property issues,909,888
42,Sexual harassment at work place,37,20
43,Harassment at work place,548,525
44,Others,3030,2878


## 16. Combining the Reporting Periods

The cleaned reporting periods are combined into a single dataset. A period column is added so that complaint categories and outcomes can be compared across time.

In [55]:
period1["period"] = "Jan-Mar 2021"
period2["period"] = "Apr 2021-Mar 2022"

combined = pd.concat(
    [period1, period2],
    ignore_index=True
)

combined

,category,old_pending,registered,total_complaints,resolved,under_proceedings,resolution_rate,proceeding_rate,share_of_complaints,period
0,Maratial problems,1877.333333,455,6087,408,5679,6.702809,93.297191,38.302290,Jan-Mar 2021
1,Pending court claims,0.000000,0,0,0,0,NaN,NaN,0.000000,Jan-Mar 2021
2,Dowry claims,0.000000,0,0,0,0,NaN,NaN,0.000000,Jan-Mar 2021
3,"Social problems, rape etc.",1185.000000,453,4008,432,3576,10.778443,89.221557,25.220237,Jan-Mar 2021
4,Property issues,287.333333,146,1008,139,869,13.789683,86.210317,6.342814,Jan-Mar 2021
5,Sexual harassment at work place,35.000000,8,113,11,102,9.734513,90.265487,0.711050,Jan-Mar 2021
6,Harassment at work place,387.333333,108,1270,119,1151,9.370079,90.629921,7.991442,Jan-Mar 2021
7,Others,1016.333333,357,3406,349,3057,10.246624,89.753376,21.432167,Jan-Mar 2021
8,Maratial problems,2374.000000,2819,5193,2841,2352,NaN,NaN,NaN,Apr 2021-Mar 2022
9,"Social problems, rape etc.",1659.000000,2803,4462,3020,1442,NaN,NaN,NaN,Apr 2021-Mar 2022


In [56]:
combined.shape

(14, 10)

In [57]:
combined[
    [
        "period",
        "category",
        "total_complaints",
        "resolved",
        "under_proceedings"
    ]
]

,period,category,total_complaints,resolved,under_proceedings
0,Jan-Mar 2021,Maratial problems,6087,408,5679
1,Jan-Mar 2021,Pending court claims,0,0,0
2,Jan-Mar 2021,Dowry claims,0,0,0
3,Jan-Mar 2021,"Social problems, rape etc.",4008,432,3576
4,Jan-Mar 2021,Property issues,1008,139,869
5,Jan-Mar 2021,Sexual harassment at work place,113,11,102
6,Jan-Mar 2021,Harassment at work place,1270,119,1151
7,Jan-Mar 2021,Others,3406,349,3057
8,Apr 2021-Mar 2022,Maratial problems,5193,2841,2352
9,Apr 2021-Mar 2022,"Social problems, rape etc.",4462,3020,1442


## 17. Resolution Rate Across Reporting Periods

To compare outcomes across reporting periods, I calculate the percentage of complaints in each category that were resolved.

In [58]:
combined["resolution_rate"] = (
    combined["resolved"] / combined["total_complaints"] * 100
)

combined[
    [
        "period",
        "category",
        "total_complaints",
        "resolved",
        "resolution_rate"
    ]
]

,period,category,total_complaints,resolved,resolution_rate
0,Jan-Mar 2021,Maratial problems,6087,408,6.702809
1,Jan-Mar 2021,Pending court claims,0,0,NaN
2,Jan-Mar 2021,Dowry claims,0,0,NaN
3,Jan-Mar 2021,"Social problems, rape etc.",4008,432,10.778443
4,Jan-Mar 2021,Property issues,1008,139,13.789683
5,Jan-Mar 2021,Sexual harassment at work place,113,11,9.734513
6,Jan-Mar 2021,Harassment at work place,1270,119,9.370079
7,Jan-Mar 2021,Others,3406,349,10.246624
8,Apr 2021-Mar 2022,Maratial problems,5193,2841,54.708261
9,Apr 2021-Mar 2022,"Social problems, rape etc.",4462,3020,67.682654


### Change in Resolution Rate

I compare the resolution rate between the two reporting periods for each complaint category to identify where the largest changes occurred.

In [59]:
rate_change = combined.pivot(
    index="category",
    columns="period",
    values="resolution_rate"
)

rate_change["change_in_percentage_points"] = (
    rate_change["Apr 2021-Mar 2022"]
    - rate_change["Jan-Mar 2021"]
)

rate_change.sort_values(
    "change_in_percentage_points",
    ascending=False
)

period,Apr 2021-Mar 2022,Jan-Mar 2021,change_in_percentage_points
category,,,
Others,70.109622,10.246624,59.862999
Property issues,72.077922,13.789683,58.288240
"Social problems, rape etc.",67.682654,10.778443,56.904210
Maratial problems,54.708261,6.702809,48.005452
Harassment at work place,55.263158,9.370079,45.893079
Sexual harassment at work place,28.571429,9.734513,18.836915
Dowry claims,NaN,NaN,NaN
Pending court claims,NaN,NaN,NaN


## 18. Key Finding

Resolution rates increased across all six complaint categories between the two reporting periods. However, sexual harassment at the workplace recorded the smallest improvement and remained the category with the lowest resolution rate in the second period.

In [60]:
period2["pending_share"] = (
    period2["under_proceedings"]
    / period2["under_proceedings"].sum()
    * 100
)

period2[
    ["category", "under_proceedings", "pending_share"]
].sort_values(
    "under_proceedings",
    ascending=False
)

,category,under_proceedings,pending_share
37,Maratial problems,2352,40.273973
40,"Social problems, rape etc.",1442,24.691781
44,Others,1227,21.010274
43,Harassment at work place,425,7.277397
41,Property issues,344,5.890411
42,Sexual harassment at work place,50,0.856164


## 19. Where Are Pending Cases Concentrated?

The distribution of complaints under proceedings shows which categories contribute most to the commission's remaining caseload. This helps identify where unresolved cases are concentrated.

In [61]:
period2[
    ["category", "under_proceedings", "pending_share"]
].sort_values(
    "pending_share",
    ascending=False
)

,category,under_proceedings,pending_share
37,Maratial problems,2352,40.273973
40,"Social problems, rape etc.",1442,24.691781
44,Others,1227,21.010274
43,Harassment at work place,425,7.277397
41,Property issues,344,5.890411
42,Sexual harassment at work place,50,0.856164


## 20. Key Findings

The analysis of the Maharashtra State Women's Commission complaint data shows three main patterns:

1. A total of 15,892 complaints were recorded in the first reporting period, of which 1,458 were resolved and 14,434 remained under proceedings.

2. In the second reporting period, 16,012 complaints were recorded, with 10,172 resolved and 5,840 under proceedings.

3. Marital problems were the largest complaint category in both periods. In the second period, they accounted for 2,352 complaints under proceedings, or about 40% of the total pending caseload.

4. Resolution rates increased across all six complaint categories between the two reporting periods. However, sexual harassment at the workplace recorded the lowest resolution rate in the second period, at 28.6%.

5. The three largest complaint categories — marital problems, social problems/rape etc., and others — accounted for about 85% of complaints in the first reporting period.

## 21. Story Pitch

### Headline
Marital Complaints Dominate Women's Commission's Pending Cases

### Pitch

Data from the Maharashtra State Women's Commission shows that marital problems were the largest category of complaints and accounted for about 40% of cases still under proceedings in the April 2021–March 2022 reporting period. While the overall resolution rate rose sharply compared with the previous reporting period, sexual harassment complaints at the workplace continued to have the lowest resolution rate at 28.6%.

The story will examine why marital complaints form such a large share of the commission's pending workload and why workplace sexual-harassment cases have a substantially lower resolution rate than other categories. Interviews with women's rights lawyers, commission officials and complainants could help explain whether the differences reflect the nature of the cases, the legal process or the commission's capacity to resolve complaints.

The data also shows that the three largest complaint categories accounted for about 85% of complaints in the first reporting period, suggesting that the commission's workload is heavily concentrated in a small number of complaint types.

## 22. Data Limitations

The source spreadsheet contains multiple reporting periods and includes cumulative complaint outcomes such as resolved complaints and complaints under proceedings. The reporting periods are also of different lengths, so raw complaint and resolution counts should not be directly compared without considering the period covered.

The analysis therefore focuses primarily on percentages and category-level patterns. The figures in the original total rows were also checked against the individual category values, and calculations in this notebook use the category-level data.

# Maharashtra State Women's Commission: Complaint Data Analysis

### A data investigation into complaint patterns and resolution rates

**Dataset:** Maharashtra State Women's Commission  
**Analysis:** Complaint categories, resolution rates and pending cases